### ISSUE TITLE
zpoline can not mprotect zero page as execute-only.

### ISSUE THREAD

**Issue by mbs0221** [yasukata/zpoline#22]
When I managed to integrate `zpoline` into our system, it can not mprotect `zero page` as execute-only, even if I changed the `mmap_min_addr` successfully.

```bash
$ LIBZPHOOK=libzphook_basic.so LD_PRELOAD=libzpoline.so ls -l
map failed
NOTE: /proc/sys/vm/mmap_min_addr should be set 0
$ sudo sh -c "echo 0 > /proc/sys/vm/mmap_min_addr"
[sudo] password for public: 
$ LIBZPHOOK=libzphook_basic.so LD_PRELOAD=libzpoline.so ls -l
ls: main.c:502: setup_trampoline: Assertion `!mprotect(0, 0x1000, PROT_EXEC)' failed.
Aborted (core dumped)
$ uname -msr
Linux 6.5.0-snp-guest-525dfdeadc3a x86_64
$
```

However, it works on another machine with the following linux version.

```bash
➜  zpoline git:(master) ✗ uname -msr
Linux 5.15.0-53-generic x86_64
```

It might be problematic on higher version of linux.

**Comment by yasukata** [comment]
Thank you for your message.

> It might be problematic on higher version of linux.

I tried the program with Linux version 6.8. However, I could not reproduce the reported issue. Therefore, I think the Linux version is not the problem.

I am guessing that there would be some security features preventing the program from updating the memory access permission. What do you think of this guess?

**Comment by mbs0221** [comment]
This problem has been solved by using linux-6.8.
